functions/video.js

```diff
@@ -1,46 +1,57 @@
 const functions = require("firebase-functions");
+const { HttpsError } = require("firebase-functions/lib/providers/https");
 
-const twilio = require("twilio");
-const AccessToken = twilio.jwt.AccessToken;
-const { VideoGrant } = AccessToken;
-
-const PROJECT_ID = functions.config().project.id;
-const TWILIO_CONFIG = functions.config().twilio;
-
-const generateToken = () => {
-  return new AccessToken(
-    TWILIO_CONFIG.account_sid,
-    TWILIO_CONFIG.api_key,
-    TWILIO_CONFIG.api_secret
-  );
-};
-
-const videoToken = (identity, room) => {
-  const videoGrant = new VideoGrant({ room });
-  const token = generateToken();
-  token.addGrant(videoGrant);
-  token.identity = identity;
-  return token;
-};
-
-exports.getToken = functions.https.onCall((data, context) => {
-  if (!context.auth || !context.auth.token) {
-    throw new functions.https.HttpsError("unauthenticated", "Please log in");
-  }
+const { RtcRole, generateAgoraTokenForAccount } = require("./src/utils/agora");
+const { assertValidAuth } = require("./src/utils/assert");
+const { twilioVideoToken } = require("./src/utils/twilio");
+
+// @debt either remove data.identity entirely, or validate that it matches the context.auth.uid
+//   (once checking that this won't break anything in the app)
+exports.getTwilioToken = functions.https.onCall((data, context) => {
+  assertValidAuth(context);
 
-  if (context.auth.token.aud !== PROJECT_ID) {
-    throw new functions.https.HttpsError("permission-denied", "Token invalid");
+  if (!data || !data.identity || !data.room) {
+    throw new HttpsError("invalid-argument", "identity or room data missing");
   }
 
-  if (data && data.identity && data.room) {
-    const token = videoToken(data.identity, data.room);
-    return {
-      token: token.toJwt(),
-    };
```

Comment:
[note] `assertValidAuth` handles the `context.auth` checks that used to be inline in this function